src/katello/agent/katelloplugin.py

```diff
@@ -162,7 +167,7 @@ def update_settings():
     # the client certs
     ca_candidates = [os.path.join(ca_cert_dir, 'katello-default-ca.pem'),
                      os.path.join(ca_cert_dir, 'candlepin-local.pem'),
-                     rhsm_conf['rhsm']['repo_ca_cert'] % rhsm_conf['rhsm']]
+                     rhsm_conf['rhsm']['repo_ca_cert'] % {'ca_cert_dir': ca_cert_dir}]
```

Comment:
This was the only line I was iffy at, as repo_ca_cert looks like `%(ca_cert_dir)skatello-server-ca.pem`
so this is doing a string substitution.  We only ever set it to use ca_cert_dir but there really isn't rule saying any other config could be used (although I don't know why any other could be used).  

The other option would be to merge the rhsm_conf['rhsm'] and rhsm_conf['server'] dicts together and pass that in as substitution.  This is simpler and I think would get the job done.